server/node/src/main/java/org/kaaproject/kaa/server/control/service/DefaultControlService.java

```diff
@@ -1015,6 +1019,56 @@ public void onNodeAdded(OperationsNodeInfo node) {
     return resolver.getNode(entityId);
   }
 
+  /**
+   * Used when writing OperationsNodeInfo instance in logs. OperationsNodeInfo instance has
+   * ByteBuffer fields which written incorrectly in the log file. In this method ByteBuffer fields
+   * temporary overwrite to null, print logs, and after ByteBuffer fields recover on theirs previous
+   * values.
+   *
+   * @param logString is a usual string for logging. Use '{}' for output OperationsNodeInfo and
+   *                  the second '{}' is OperationsServerResolver
+   * @param node      in which find and temporary overwrite all ByteBuffer-s to null
+   * @param resolver  OperationsServerResolver instance
+   */
+  private void writeLogWithoutByteBuffer(String logString,
+                                         OperationsNodeInfo node,
+                                         OperationsServerResolver resolver) {
+    //temporary set connection info of transports to null
+    Map<TransportMetaData, Map<VersionConnectionInfoPair, ByteBuffer>> map = new HashMap<>();
+    HashMap<VersionConnectionInfoPair, ByteBuffer> innerMap;
```

Comment:
Move type declaration under for statement because you never use one after the loop.